### PR TITLE
Alerting: Create integration testing infra for recording rules

### PR DESCRIPTION
**Pull request by alexweav**
**What is this feature?**

Adds tooling to improve our integration testing for recording rules.

- We now stably produce a query that can either execute or fail depending on the test, where before the query was random.
  - This means we can deterministically assert on success or failure behavior, rather than only asserting that an eval happened.
- Creates an httptest-based remote write target. You can point a real writer to it, intercept and count requests, and compare behavior.

Includes an integration test for queries that error. The same test for the Success case, revealed an edge case where we do not correctly handle dataframes with no metadata field at all, will be fixed in a future PR.

**Which issue(s) does this PR fix?**:

n/a

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [x] If this is a pre-GA feature, it is behind a feature toggle.
- [x] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/services/ngalert/schedule/recording_rule_test.go [hunk 1]
```diff
@@ -156,64 +162,146 @@ func TestRecordingRule_Integration(t *testing.T) {
...
+			"expression":"2 + 1"
+		}`
+	case "error":
+		expression = `{
+			"datasourceUid": "__expr__",
+			"type":"math",
+			"expression":"$NOTEXIST"
+		}`
+	default:
+		require.Fail(t, "Query generation for health %s is not supported yet", health)
```
Comment (yuri-tceretian) [inline pkg/services/ngalert/schedule/recording_rule_test.go:276]: I wonder if it is just good enough to panic? or even just empty string, which hopefully will cause tests to fail :) having testing.T in mutator function seems strange.

**Review by yuri-tceretian — APPROVED** [review]

File: pkg/services/ngalert/schedule/recording_rule_test.go (hunk 1 above)
Comment (alexweav) [reply pkg/services/ngalert/schedule/recording_rule_test.go:276]: Yeah agreed, i like that more :+1:
